server/api/impl/upstream/uphttp/upstream.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2020-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package uphttp
```

Comment:
This package will be implemented by the aws connector right? Basically my PR has it already implemented(kinda), correct? 
Or are we abstracting away different upstream mechanisms and we will keep this as well as aws one?

Reply:
Abstracting away, we need a new package `upawslambda` that would implement `api.Upstream` for Lambda. We want to keep the `uphttp` practically as is to support remote Apps. I am also keen on the use of "builtIn" for the -apps plugin itself, and perhaps some Apps that we can migrate from go plugins to "built-in" Apps easily.

Reply:
"built-in" apps will be same as plugins but with extra layers :)
Understood